app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt

```diff
@@ -0,0 +1,254 @@
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun verifyContextOpenLinkNewTab() {
+        val pageLinks =
+            TestAssetHelper.getGenericAsset(mockWebServer, 4)
+        val genericURL =
+            TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(pageLinks.url) {
+            verifyPageContent(pageLinks.content)
+            longClickMatchingText("Link 1")
+            verifyLinkContextMenuItems(genericURL.url)
+            clickContextOpenLinkInNewTab()
+            verifySnackBarText("New tab opened")
+            snackBarButtonClick("Switch")
+            verifyUrl(genericURL.url.toString())
+        }.openHomeScreen {
+            verifyHomeScreen()
+            verifyExistingOpenTabs("Test_Page_1")
+            verifyExistingOpenTabs("Test_Page_4")
+        }
+    }
+
+    @Test
+    fun verifyContextOpenLinkPrivateTab() {
+        val pageLinks =
+            TestAssetHelper.getGenericAsset(mockWebServer, 4)
+        val genericURL =
+            TestAssetHelper.getGenericAsset(mockWebServer, 2)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(pageLinks.url) {
+            verifyPageContent(pageLinks.content)
+            longClickMatchingText("Link 2")
+            verifyLinkContextMenuItems(genericURL.url)
+            clickContextOpenLinkInPrivateTab()
+            verifySnackBarText("New private tab opened")
+            snackBarButtonClick("Switch")
+            verifyUrl(genericURL.url.toString())
+        }.openHomeScreen {
+            verifyPrivateSessionHeader()
+            verifyExistingOpenTabs("Test_Page_2")
+        }
+    }
+
+    @Test
+    fun verifyContextCopyLink() {
+        val pageLinks =
+            TestAssetHelper.getGenericAsset(mockWebServer, 4)
+        val genericURL =
+            TestAssetHelper.getGenericAsset(mockWebServer, 3)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(pageLinks.url) {
+            verifyPageContent(pageLinks.content)
+            longClickMatchingText("Link 3")
+            verifyLinkContextMenuItems(genericURL.url)
+            clickContextCopyLink()
+            verifySnackBarText("Link copied to clipboard")
+        }.openNavigationToolbar {
+        }.visitLinkFromClipboard(genericURL.url) {
+            verifyUrl(genericURL.url.toString())
+        }
+    }
+
+    @Test
+    fun verifyContextShareLink() {
+        val pageLinks =
+            TestAssetHelper.getGenericAsset(mockWebServer, 4)
+        val genericURL =
+            TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(pageLinks.url) {
+            verifyPageContent(pageLinks.content)
+            longClickMatchingText("Link 1")
+            verifyLinkContextMenuItems(genericURL.url)
+            clickContextShareLink()
+            // verify share intent is matched with associated type
+            intended(
```

Comment:
Can this be moved into a method within the browser robot or maybe the test helper? This way the readability of the test stays intact.

Reply:
Moved into the associated robot.